components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -246,43 +245,34 @@ internal class ConstellationObserver(
     }
 
     internal fun onSubscribeError(e: Exception) {
-        logger.warn("Re-subscribing failed; FxA push events will not be received.", e)
-        breadcrumb(
-            exception = e,
-            message = "Re-subscribing to failed FxA push after subscriptionExpired"
-        )
-    }
+        val errorMessage = "Re-subscribing failed; FxA push events will not be received."
 
-    internal fun onUnsubscribeError(e: Exception) {
-        logger.warn("Un-subscribing failed; subscribe may return the same endpoint", e)
-        breadcrumb(
-            exception = e,
-            message = "Un-subscribing to failed FxA push after subscriptionExpired"
-        )
+        logger.warn(errorMessage, e)
+        crashReporter?.submitCaughtException(IllegalStateException(errorMessage, e))
```

Comment:
I suggest using a specific exception class here, it'll be much easier to filter-for in sentry and soccorro afterwards. (ISE's are easy to miss/loose sight of)

Reply:
Also: are you able to tell that this happened because of a harmless error (e.g. network failure)? Does it matter, even?

In other words, what's the goal here - to judge if we need a recovery path here (to handle transient errors)? To understand if unexpected things we don't know about can happen?

I guarantee you without even looking at the volumes here that this _will_ sometimes fail, and that you do need to have a recovery path if you want to cover your bases fully. However, we need to be able to tell apart recoverable cases from really abnormal things that we should actually look at.

Reply:
This already fails with an error which we've filed https://github.com/mozilla-services/autopush/issues/1445 for it. This more for tracking the number of affected users at this point. 🙂 

> I suggest using a specific exception class here, it'll be much easier to filter-for in sentry and soccorro afterwards. 

I've been using the error message which has worked, but I think this is better.

Reply:
Too clarify, as we get the fixes we need, this would be a good graph to see the errors reduce. A gauge of Send Tab with Push failures.